firefox-ios/Client/Frontend/Browser/Toolbars/Redux/NavigationBarState.swift

```diff
@@ -30,75 +30,91 @@ struct NavigationBarState: StateType, Equatable {
 
         switch action.actionType {
         case ToolbarActionType.didLoadToolbars:
-            guard let model = (action as? ToolbarAction)?.navigationToolbarModel else { return state }
-
-            return NavigationBarState(
-                windowUUID: state.windowUUID,
-                actions: model.actions ?? state.actions,
-                displayBorder: model.displayBorder ?? state.displayBorder
-            )
+            return handleToolbarDidLoadToolbars(state: state, action: action)
 
         case ToolbarActionType.numberOfTabsChanged:
-            guard let numberOfTabs = (action as? ToolbarAction)?.numberOfTabs else { return state }
+            return handleToolbarNumberOfTabsChanged(state: state, action: action)
 
-            var actions = state.actions
+        case ToolbarActionType.backButtonStateChanged:
+            return handleToolbarBackButtonStateChanged(state: state, action: action)
 
-            if let index = actions.firstIndex(where: { $0.actionType == .tabs }) {
-                actions[index].numberOfTabs = numberOfTabs
-            }
+        case ToolbarActionType.forwardButtonStateChanged:
+            return handleToolbarForwardButtonStateChanged(state: state, action: action)
 
-            return NavigationBarState(
-                windowUUID: state.windowUUID,
-                actions: actions,
-                displayBorder: state.displayBorder
-            )
+        case ToolbarActionType.scrollOffsetChanged:
+            return handleToolbarScrollOffsetChanged(state: state, action: action)
 
-        case ToolbarActionType.backButtonStateChanged:
-            guard let isEnabled = (action as? ToolbarAction)?.isButtonEnabled else { return state }
+        default:
+            return state
```

Comment:
In the previous code, the `struct` was recreated with the same values. I simply returned it directly because the result is the same. Does anyone have a different perspective?